.github/workflows/tests-ipsec-upgrade.yaml

```diff
@@ -360,12 +377,26 @@ jobs:
           kubectl get pods --all-namespaces -o wide
           kubectl -n kube-system exec daemonset/cilium -c cilium-agent -- cilium-dbg status
 
-      - name: Check conn-disrupt-test after downgrading
+      - name: Run connection interrupted tests after downgrading (${{ join(matrix.*, ', ') }})
+        if: ${{ steps.vars.outputs.downgrade_version != '' }}
+        uses: ./.github/actions/conn-disrupt-test-check
+        with:
+          job-name: cilium-downgrade-${{ matrix.name }}
```

Comment:
Could you explain in the commit description why this change is needed? I think the `seq-` / `!seq-` are likely self-explanatory for most, but it's harder to guess why the conn-disrupt test wouldn't be parallelizable.

Reply:
Honestly there is no special reason. It was suggested by Martynas that we should split the steps to make it more obvious that the conn-disrupt tests are an "important" part to check after an upgrade / downgrade. This was actually not obvious when I was adding concurrency to the e2e-upgrade tests as I got confused which tests were checking for the conn-disrupt and which tests weren't. Thus, I just applied the same logic in here.

Now, if you ask why aren't we running the `conn-disrupt` in parallel? I would say it's simply because there's only, as far as I know, one test executed and therefore it doesn't make sense to run in parallel.

Reply:
Ah. So we still don't know why https://github.com/cilium/cilium/pull/35362#issuecomment-2419354256 happened?

Reply:
I haven't seen it happening since I rebased. But your explanation in the comment would make more sense indeed.